lib/api/core/storage/models/baseModel.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+class BaseModel {
+
+  static register (ModelClass, collection) {
+    ModelClass.collection = collection;
+
+    ModelClass.load = async id => {
+      const { _source, _id } = await BaseModel.indexStorage.get(
+        ModelClass.collection,
+        id);
+
+      return new ModelClass({ _id, _source });
+    };
+
+    ModelClass.deleteByQuery = searchBody =>
+      BaseModel.indexStorage.deleteByQuery(ModelClass.collection, searchBody);
+
+    ModelClass.search = async (searchBody, options) => {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a semicolon at the end of this statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ASemicolon)